source/base/DateTime.ooc

```diff
@@ -14,22 +14,246 @@
+	}
+
+	toString: func -> String {
+		this toStringFormat(This DefaultFormat)
+	}
+	// <summary>
+	// Convert this object to string representation
+	// </summary>
+	// supported formatting expressions:
+	//	%yyyy - year
+	//	%yy	- year as two digit number
+	//  %MM - month with leading zero
+	//	%M	- month without leading zero
+	//	%dd - day with leading zero
+	//	%d	- day without leading zero
+	//	%hh - hour
+	//	%mm - minute
+	//	%ss - second
+	//	%zzzz - millisecond
+	// <param name="format">output format specification</param>
+	toStringFormat: func (format: String) -> String {
+		result := format
+		data := This _ticksToDateTimeHelper(this ticks)
+		result = result replaceAll("%yyyy", "%d" format(data year))
+		result = result replaceAll("%yy", "%d" format(data year % 100))
+		result = result replaceAll("%MM", (data month < 10 ? "0%d" : "%d") format(data month))
+		result = result replaceAll("%dd", (data day < 10 ? "0%d" : "%d") format(data day))
+		result = result replaceAll("%M", "%d" format(data month))
+		result = result replaceAll("%d", "%d" format(data day))
+		result = result replaceAll("%hh", "%d" format(data hour))
+		result = result replaceAll("%mm", "%d" format(data minute))
+		result = result replaceAll("%ss", "%d" format(data second))
+		result = result replaceAll("%zzzz", "%d" format(data millisecond))
+		result
+	}
+
+	compareTo: func (other: This) -> Order {
+		if (this ticks > other ticks)
+			Order greater
+		else if (this ticks < other ticks)
+			Order less
+		else
+			Order equal
+	}
+
+	operator - (other: This) -> TimeSpan {
+		TimeSpan new(this ticks as Int64 - other ticks as Int64)
+	}
+	operator + (span: TimeSpan) -> This {
+		This new(this ticks as Int64 + span ticks)
+	}
+	operator - (span: TimeSpan) -> This {
+		This new(this ticks as Int64 - span ticks)
+	}
+	operator == (other: This) -> Bool {
+		this compareTo(other) == Order equal
+	}
+	operator != (other: This) -> Bool {
+		this compareTo(other) != Order equal
+	}
+	operator < (other: This) -> Bool {
+		this compareTo(other) == Order less
+	}
+	operator <= (other: This) -> Bool {
+		this compareTo(other) != Order greater
+	}
+	operator > (other: This) -> Bool {
+		this compareTo(other) == Order greater
+	}
+	operator >= (other: This) -> Bool {
+		this compareTo(other) != Order less
+	}
+
+	/* number of days in year ( non-leap ) */
+	DaysInYear: static const Int = 365
+	DaysInFourYears: static const Int = 3 * 365 + 366
+	TicksPerMillisecond: static const Int64 = 1000
+	TicksPerSecond: static const Int64 = TicksPerMillisecond * 1000
+	TicksPerMinute: static const Int64 = TicksPerSecond * 60
+	TicksPerHour: static const Int64 = TicksPerMinute * 60
+	TicksPerDay: static const Int64 = TicksPerHour * 24
+	TicksPerWeek: static const UInt64 = TicksPerDay * 7
+	TicksPerFourYears: static const Int64 = DaysInFourYears * TicksPerDay
+	/* default date/time printing format */
+	DefaultFormat: static const String = "%yyyy-%MM-%dd %hh:%mm:%ss::%zzzz"
```

Comment:
For no particular reason, constants like these are not named with a capital in the rest of the code, so they shouldn't be that way here, either.